src/common/ServerSocket.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+#include <netinet/in.h>
+#include <memory>
+#include "Socket.h"
+
+class ServerSocket
+{
+public:
+    explicit ServerSocket(int socket_fd);
+
+    ~ServerSocket();
+
+    void bind(const sockaddr_in &server_address) const;
```

Comment:
dont use raw system api (linux api) in ur code (API), always create a wrapping around them and at the final point convert \ cast \ open \ redirect to the raw types